composition/rental.rb

```diff
@@ -0,0 +1,29 @@
+require 'securerandom'
+require_relative 'book'
+require_relative '../person'
+
+class Rental
+  attr_reader :id, :book, :person
+  attr_accessor :date
+
+  def initialize(date, book, person)
+    @id = SecureRandom.hex
+    @date = date
+
+    @book = book
+    book.rentals << self
+
+    @person = person
+    person.rentals << self
+  end
+
+  def book=(book)
+    @book = book
+    book.rentals.push(self) unless book.rentals.include?(self)
+  end
+
+  def person=(person)
+    @person = person
+    person.rentals.push(self) unless person.rentals.include?(self)
+  end
```

Comment:
- [ ] Great work creating these codes, I can see that you are manually creating the _**setter**_ methods instead of using the `attr_accessor` method. Kindly remove this line of codes from each setter method because you have them written in the `initialize` method of this class:

```ruby
 person.rentals.push(self) unless person.rentals.include?(self)
```
```ruby
 book.rentals.push(self) unless book.rentals.include?(self)
```